packages/api/cms-api/src/warnings/warning.resolver.ts

```diff
@@ -25,10 +30,73 @@ export class WarningResolver {
     }
 
     @Query(() => PaginatedWarnings)
-    async warnings(@Args() { status, search, filter, sort, offset, limit }: WarningsArgs): Promise<PaginatedWarnings> {
-        const where = gqlArgsToMikroOrmQuery({ search, filter }, this.repository);
+    async warnings(
+        @Args() { status, search, filter, scopes, sort, offset, limit }: WarningsArgs,
+        @GetCurrentUser() user: CurrentUser,
+    ): Promise<PaginatedWarnings> {
+        // check if there are any scopes that the user does not have permission to
+        const allowedScopesForUser = user.permissions.find(({ permission }) => permission === "warnings")?.contentScopes;
+
+        for (const scope of scopes) {
```

Comment:
Out of scope: Should the user permissions module supports a magic `scopes` (like it does for `scope`) per default?

Reply:
I think it would be handy, but I'm not that familiar with the user permissions module

Reply:
@fraxachun @nsams what do you think?

Reply:
This one: https://github.com/vivid-planet/comet/blob/main/packages/api/cms-api/src/user-permissions/content-scope.service.ts#L86.